tests/Integration/Behaviour/Features/Context/Domain/Product/UpdatePackFeatureContext.php

```diff
@@ -84,6 +85,20 @@ public function removeAllProductsFromPack(string $packReference): void
         }
     }
 
+    /**
+     * @When pack :packReference should be empty
+     *
+     * @param string $packReference
+     */
+    public function assertPackEmpty(string $packReference): void
+    {
+        $packId = $this->getSharedStorage()->get($packReference);
+
+        $packedProducts = $this->getQueryBus()->handle(new GetPackedProducts($packId));
+        Assert::assertEmpty($packedProducts);
+        Assert::assertFalse(Pack::isPack($packId));
```

Comment:
How a pack cannot be a pack ?

Reply:
Because `Pack::isPack` is a method that returns the presence of pack associations Up until now we only relied on the presence/absence of this associations to define the type of the product

That's the whole point of the PR actually (explained in the description) we do not rely only on associations any more but on a field that represents (and persist) the product type

TLDR; when is a pack not a pack? When its product_type field is equal to pack but the merchant has not defined associations with other products yet

Reply:
Or when you just removed all its products, then no association but the `product_type` is still pack